prow/scripts/cluster-integration/kyma-gke-upgrade.sh

```diff
@@ -344,40 +346,59 @@ function checkTestPodTerminated() {
     return 1
 }
 
-createTestResources() {
-    shout "Create e2e upgrade test resources"
-    date
+function installTestChartOrFail() {
+  local path=$1
+  local name=$2
+  local namespace=$3
 
-    injectTestingAddons
+  shout "Create ${name} resources"
+  date
 
-    if [  -f "$(helm home)/ca.pem" ]; then
-        local HELM_ARGS="--tls"
-    fi
+  local HELM_ARGS
+  if [[ -f "$(helm home)/ca.pem" ]]; then
+      HELM_ARGS="--tls"
+  fi
+
+  helm install "${path}" \
+      --name "${name}" \
+      --namespace "${namespace}" \
+      --timeout "${HELM_TIMEOUT_SEC}" \
+      --set domain="${DOMAIN}" \
+      --wait ${HELM_ARGS}
+
+  prepareResult=$?
+  if [[ "${prepareResult}" != 0 ]]; then
+      echo "Helm install ${name} operation failed: ${prepareResult}"
+      exit "${prepareResult}"
+  fi
+}
 
-    helm install "${UPGRADE_TEST_PATH}" \
-        --name "${UPGRADE_TEST_RELEASE_NAME}" \
-        --namespace "${UPGRADE_TEST_NAMESPACE}" \
-        --timeout "${UPGRADE_TEST_HELM_TIMEOUT_SEC}" \
-        --wait ${HELM_ARGS}
+function waitForTestPodToFinish() {
```

Comment:
Another candidate for `library.sh`, I guess or can be scooped out to be an one-thing-do-well script inside `test-infra/prow/scripts`

Reply:
As discussed offline, we will keep it like that for now.